docs/paper/dev/misc/internal-code.mdx

```diff
@@ -0,0 +1,79 @@
+---
+slug: /dev/internals
+---
+
+# Minecraft Internals
+
+The code that runs Minecraft is not open source. Bukkit is an API that allows plugins to interact with the server. This
+is implemented by CraftBukkit and interacts with the Minecraft code. You will often hear the terms NMS and CraftBukkit
+when talking about Minecraft internals. NMS stands for `net.minecraft.server` and refers to a Java package that contains
+a lot of Mojang's code.
+
+:::danger[Using Minecraft Internals]
+
+Using Minecraft internals is not recommended. This is because the using the internals directly is not guaranteed to be
+stable and they change often. This means that your plugin may break when a new version of Minecraft is released.
+
+**PaperMC will offer no direct support for programming against Minecraft internals.**
+
+:::
+
+## Accessing Minecraft Internals
+
+In order to use Mojang and CraftBukkit code, you may either use the Paperweight Userdev plugin or use reflection.
+We would recommend using the [Paperweight Userdev](https://github.com/PaperMC/paperweight-test-plugin)
```

Comment:
```suggestion
We recommend using the [Paperweight Userdev](https://github.com/PaperMC/paperweight-test-plugin)
```
maybe?

Reply:
It shouldn't say `We` either I think. `Using pw userdev is recommended....`